doc/mavlink_gitbook.py

```diff
@@ -40,17 +40,48 @@
 index_text="""<!-- THIS FILE IS AUTO-GENERATED (DO NOT UPDATE GITBOOK): https://github.com/mavlink/mavlink/blob/master/doc/mavlink_gitbook.py -->
 # Dialects {#dialects}
 
-MAVLink *dialects* are XML files that define *protocol-* and *vendor-specific* messages, enums and commands.
+MAVLink *dialects* are XML definition files that define *protocol-* and *vendor-specific* messages, enums and commands.
 
-Dialects may *include* other MAVLink XML files.
+Dialects may *include* other MAVLink XML files, which may in turn contain other XML files (up to 5 levels of XML file nesting are allowed - see `MAXIMUM_INCLUDE_FILE_NESTING` in [mavgen.py](https://github.com/ArduPilot/pymavlink/blob/master/generator/mavgen.py#L44)).
 A typical pattern is for a dialect to include [common.xml](../messages/common.md) (containing the *MAVLink standard definitions*), extending it with vendor or protocol specific messages.
-While a dialect can include any other message definition, only a single level of nesting is supported ([at time of writing](https://github.com/ArduPilot/pymavlink/pull/248)).
+
+## Standard Definitions
+
+The following XML definition files are considered standard/core (i.e. not dialects):
+
+* [minimal.xml](minimal.md) - the minimum set of entities (messages, enums, MAV_CMD) required to set up a MAVLink network.
+* [standard.xml](standard.md) - the standard set of entities that are implemented by almost all flight stacks (at least 2, in a compatible way).
+  This `includes` [minimal.xml](minimal.md).
+* [common.xml](../messages/common.md) - the set of entitites that have been implemented in at least one core flight stack.
+  This `includes` [standard.xml](minimal.md)
+
+Further, [all.xml](all.md) is a _special case_.
+It includes almost all other XML definition files, and can be used to verify that there are no ID clashes (and can potentially be used by GCS to communicate with any core dialect).
+
+> **Note** We are still working towards moving the truly standard entities from **common.xml** to **standard.xml**
+  Currently you should include [common.xml](../messages/common.md)
+
+## Core Dialects
+
+Core dialects are stored in [mavlink/message definitions](https://github.com/mavlink/mavlink/blob/master/message_definitions/).
+These are the dialects for the major MAVLink stakeholder flight stacks.
 
 > **Note** Vendor forks of MAVLink may contain dialect messages that are not yet merged, and hence will not appear in this documentation.
 
-The dialect files are stored alongside in separate XML files in [mavlink/message definitions](https://github.com/mavlink/mavlink/blob/master/message_definitions/).
+Human-readable forms of all the the core dialects are linked below:
+"""
+
+index_text_trailer="""## External Dialects
+
+MAVLink provides the [/external/dialects](https://github.com/mavlink/mavlink/tree/master/external/dialects) folder for dialects from projects that are not maintained by core MAVLink stakeholders or part of the MAVLink standard.
```

Comment:
We don't provide this anymore. It was removed.

Reply:
I got this patch from upstream. It needs to be removed from upstream then.

Reply:
Done. Fixed by https://github.com/mavlink/mavlink/pull/1802